celery/contrib/testing/worker.py

```diff
@@ -121,16 +124,26 @@ def _start_worker_thread(app,
         without_gossip=True,
         **kwargs)
 
+    # TODO: Shall we consider starting the thread with daemon=True
     t = threading.Thread(target=worker.start)
     t.start()
     worker.ensure_started()
     _set_task_join_will_block(False)
 
     yield worker
 
+    # TODO: Shall we check if the worker is still executing tasks, so that
+    # the timeout below is only applied to the shutdown operation?
```

Comment:
We don't need to check that. Timeout of `join()` is just upper limit. If worker is not processing tasks, `join()` should return immediately.